andela/travel_tool_back
# #161646967 Fix yarn version inconsistency

#### What does this PR do?
 - Introduces a check for the yarn version being used to perform yarn commands
 - Modifies Dockerfile and circle CI to install version 1.12.x of yarn.

#### Description of Task to be completed?
- Introduce a check for the yarn version being used to perform yarn commands
- Modifiy Dockerfile and circle CI to install version 1.12.x of yarn.

#### How should this be manually tested?
 - Clone the repo
    `git clone https://github.com/andela/travel_tool_back.git`
 - Switch to the project directory
   `cd travel_tool_back`
 - Checkout the **fix-yarn-lock-inconsistencies-161775747** branch
   `git checkout fix-yarn-lock-inconsistencies-161775747`
 - Using `yarn`:
    - Run `yarn install` to install dependencies.
       It is expected that the yarn.lock file will not be modified and that the output on the terminal will look close to this:
![image](https://user-images.githubusercontent.com/41262970/48265843-ad377c00-e43e-11e8-851d-712f333960b1.png)

       Please note that `Saved lockfile` is not printed on the terminal.
       Please note that the yarn command is run with `yarn v1.12.3`.

     - Run `yarn check --integrity`
It is expected that the check succeeds and that the output in the terminal will look close to this:
![image](https://user-images.githubusercontent.com/41262970/48265856-b294c680-e43e-11e8-9c60-a81b017482d0.png)

     - Run `yarn start:dev`
        It is expected that the backend starts normally with no errors.

- Using `make`:
    - Run `make start`
       It is expected that the steps 1 through 9 run without fail.
       Please note that `Saved lockfile` is not printed to the terminal.
       Please note that the yarn command is run with `yarn v1.12.3`.

    - Run `make seed`
       It is expected that the database is seeded without fail.

    - Run `make test`
       It is expected that the tests run without fail.

    - Run `make ssh`.
       This will let you access the terminal of the docker environment where your app is running.
       - Inside this environment, run `yarn check --integrity`
          It is expected that the check succeeds and that the output in the terminal will look close to this:
![image](https://user-images.githubusercontent.com/41262970/48265873-bf191f00-e43e-11e8-8bb5-63df571b7064.png)

#### Any background context you want to provide?
* The **Using `make`** part of testing this PR needs one to have already installed docker.
* You will need yarn version 1.12.x to perform the yarn tests mentioned above, otherwise an error is raised as shown below:
  ![image](https://user-images.githubusercontent.com/41262970/48263383-fbe11800-e436-11e8-9ca5-56a7ea038a19.png)

#### What are the relevant pivotal tracker stories?
- [#161646967](https://www.pivotaltracker.com/story/show/161775747)

#### Questions:
N/A
